### PR TITLE
Change Console Status PublicHostname to ConsoleURL

### PR DESCRIPTION
**Pull request by benjaminapetersen**

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
@openshift/api-reviewers
@spadgett @enj

File: config/v1/types_console.go [hunk 1]
```diff
@@ -24,9 +24,9 @@ type ConsoleSpec struct {
 }
 
 type ConsoleStatus struct {
-	// The hostname for the console. This will match the host for the route that
+	// The URL for the console. This will be derived from the host for the route that
 	// is created for the console.
-	PublicHostname string `json:"publicHostname"`
```
Comment (enj) [inline config/v1/types_console.go]: json is wrong

Comment (benjaminapetersen) [reply config/v1/types_console.go]: thanks, will update.

**Comment by enj** [comment]
The only similar field in config v1:

https://github.com/openshift/api/blob/573fabd050d7895c3b76249efd428af19202e74d/config/v1/types_infrastructure.go#L44-L47

**Comment by spadgett** [comment]
+1 to this change. It makes that the scheme, host, port, and path explicit and would also give us freedom to someday support a different port on add a context root if we ever needed to.

I don't have a strong opinion on the property name. `publicURL` seems fine, but will defer to @enj and @derekwaynecarr

File: config/v1/types_console.go [hunk 2]
```diff
@@ -24,9 +24,9 @@ type ConsoleSpec struct {
...
+	PublicURL string `json:"publicURL"`
```
Comment (spadgett) [inline config/v1/types_console.go]: Maybe `consoleURL` in case we ever have another URL we publish in status?

Comment (benjaminapetersen) [reply config/v1/types_console.go]:
```yaml
apiVersion: config.openshift.io/v1
kind: Console
metadata:
  name: cluster
spec:
  authentication:
    logoutRedirect: ""
status:
  # default url for config for console could be obvious its for the console
  publicURL: ""
  # or make it explicit 
  consoleURL: ""
  # for when we add others later
  otherURL: ""
  yetAnotherURL: ""
```
I don't personally have a strong opinion.

**Comment by derekwaynecarr** [comment]
/approve

the installer should probably be updated to read from this and not the route.

File: config/v1/types_console.go (hunk 2 above)
Comment (benjaminapetersen) [reply config/v1/types_console.go]: `consoleURL` it is, updating.

**Comment by benjaminapetersen** [comment]
Updated

**Comment by spadgett** [comment]
> the installer should probably be updated to read from this and not the route.

@sallyom fyi

**Pull request merged**